src/init.rs

```diff
@@ -12,13 +12,52 @@
 // OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 // CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"),
+    target_os = "linux"))]
+use cpu_feature::arm_linux::auxv;
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"),
+    target_os = "linux"))]
+use cpu_feature::arm_linux;
+
 #[inline(always)]
 pub fn init_once() {
     #[cfg(not(all(target_arch = "aarch64", target_os = "ios")))]
     {
         extern crate std;
-        extern { fn GFp_cpuid_setup(); }
         static INIT: std::sync::Once = std::sync::ONCE_INIT;
-        INIT.call_once(|| unsafe { GFp_cpuid_setup() });
+        INIT.call_once(|| set_cpu_features() );
     }
 }
+
+// On arm linux, use pure rust
+
+#[cfg(all(any(target_arch = "arm", target_arch = "aarch64"),
+    target_os = "linux"))]
+fn set_cpu_features() {
+    let getauxval = auxv::NativeProvider {};
+
+    unsafe {
+        GFp_armcap_P |= arm_linux::armcap_from_features(getauxval);
```

Comment:
The original C used `|=` so I stuck with it, but I don't know of any other code that writes to that global.